megamek/src/megamek/client/ui/swing/ClientGUI.java

```diff
@@ -1006,12 +1020,6 @@ void die() {
         }
         client.die();
 
-        // TODO Is there a better solution?
```

Comment:
I guess there was a better solution. What happens now?

Reply:
From the former comment:
// This is required because the ChatLounge adds the listener to the
// MechSummaryCache that must be removed explicitly.
I think the call to `((Distractable) component).removeAllListeners();` a few lines above already removes that listener.